utils/cfutils.go

```diff
@@ -0,0 +1,222 @@
+package utils
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/go-yaml/yaml"
+)
+
+	
+
+type CFAppEnv struct {
+	SystemProvided interface{} `yaml:"System-Provided"`
+	VCAPSERVICES struct {
+		FsStorage []struct {
+			BindingGUID string `yaml:"binding_guid"`
+			BindingName interface{} `yaml:"binding_name"`
+			Credentials struct {
+			} `yaml:"credentials"`
+			InstanceGUID string `yaml:"instance_guid"`
+			InstanceName string `yaml:"instance_name"`
+			Label string `yaml:"label"`
+			Name string `yaml:"name"`
+			Plan string `yaml:"plan"`
+			Provider interface{} `yaml:"provider"`
+			SyslogDrainURL interface{} `yaml:"syslog_drain_url"`
+			Tags []string `yaml:"tags"`
+			VolumeMounts []struct {
+				ContainerDir string `yaml:"container_dir"`
+				DeviceType string `yaml:"device_type"`
+				Mode string `yaml:"mode"`
+			} `yaml:"volume_mounts"`
+		} `yaml:"fs-storage"`
+	} `yaml:"VCAP_SERVICES"`
+	VCAPAPPLICATION struct {
+		ApplicationID string `yaml:"application_id"`
+		ApplicationName string `yaml:"application_name"`
+		ApplicationUris []string `yaml:"application_uris"`
+		CfAPI string `yaml:"cf_api"`
+		Limits struct {
+			Fds int `yaml:"fds"`
+		} `yaml:"limits"`
+		Name string `yaml:"name"`
+		OrganizationID string `yaml:"organization_id"`
+		OrganizationName string `yaml:"organization_name"`
+		SpaceID string `yaml:"space_id"`
+		SpaceName string `yaml:"space_name"`
+		Uris []string `yaml:"uris"`
+		Users interface{} `yaml:"users"`
+	} `yaml:"VCAP_APPLICATION"`
+	UserProvided interface{} `yaml:"User-Provided"`
+	JBPCONFIGOPENJDKJRE struct {
+		Jre struct {
+			Version string `yaml:"version"`
+		} `yaml:"jre"`
+	} `yaml:"JBP_CONFIG_OPEN_JDK_JRE"`
+	JBPCONFIGSPRINGAUTORECONFIGURATION struct {
+		Enabled bool `yaml:"enabled"`
+	} `yaml:"JBP_CONFIG_SPRING_AUTO_RECONFIGURATION"`
+	RunningEnvironmentVariableGroups interface{} `yaml:"Running Environment Variable Groups"`
+	CREDHUBAPI string `yaml:"CREDHUB_API"`
+}
+
+func readAppEnv(app string) ([]byte, error){
+	env, err := exec.Command("cf", "env", app).Output()
```

Comment:
I think we should try to avoid parsing (potentially fragile) text output of the CF CLI.
The CF CLI supports multiple languages and we would need to deal with that, too.
Instead we should use `cf curl` instead and work with the returned JSON value.
You can find out which API calls are behind a command (e.g., cf env) by enabling tracing of the CF CLI and running the command again:

```shell
export CF_TRACE=true
cf env APP_NAME
```

For example, to get the actual JSON response from the cf env command, you can use:

```
cf curl /v3/apps/<APP_GUID>/env
```

Reply:
Thanks for the comments and hint, helped me understanding it a lot.